### PR TITLE
Property filter simplify types

### PR DESCRIPTION
**Pull request by pan-kot**
### Description

A refactoring to property filter that simplifies TS annotations, replaces unnecessary null-checks with default values and moves focus-no-open logic to the internal autosuggest.

### How has this been tested?

Small updates to existing unit tests. Manual checks.

### Documentation changes

[*Do the changes include any API documentation changes?*]
- [x] _Yes, this change contains documentation changes._
- [ ] _No._

Minor changes: default values applied to filteringOptions and customGroupsText.

### Related Links

[*Attach any related links/pull request for this change*]

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- [ ] _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](CONTRIBUTING.md#public-apis)._
- [ ] _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](CONTRIBUTING.md#browsers-support)._
- [ ] _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- [ ] _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- [ ] _Changes are covered with new/existing unit tests?_
- [ ] _Changes are covered with new/existing integration tests?_
</details>

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/property-filter/controller.ts [hunk 1]
```diff
@@ -1,21 +1,26 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
-import { PropertyFilterProps } from './interfaces';
-import { fireNonCancelableEvent } from '../internal/events';
+
+import { PropertyFilterProps as I } from './interfaces';
```
Comment (johannes-weber) [inline src/property-filter/controller.ts:4]: Suggestion: use the full name instead the shortened one or a more descriptive alias.

I think shortening the interface name to a single character makes reading the code harder.

File: src/property-filter/token-editor.tsx [hunk 2]
```diff
@@ -7,12 +7,11 @@ import { SelectProps } from '../select/interfaces';
 import InternalSelect from '../select/internal';
 import InternalSpaceBetween from '../space-between/internal';
 import InternalAutosuggest from '../autosuggest/internal';
-import { AutosuggestProps } from '../autosuggest/interfaces';
 import InternalPopover, { InternalPopoverRef } from '../popover/internal';
 import { InternalButton } from '../button/internal';
 import { useUniqueId } from '../internal/hooks/use-unique-id/index';
 
-import { PropertyFilterProps } from './interfaces';
+import { PropertyFilterProps as I } from './interfaces';
```
Comment (johannes-weber) [inline src/property-filter/token-editor.tsx:14]: Same as above.

**Review by johannes-weber — CHANGES_REQUESTED** [review]

File: src/property-filter/controller.ts (hunk 1 above)
Comment (pan-kot) [reply src/property-filter/controller.ts:4]: I would prefer to omit the namespace name completely when using PropertyFilterProps in property filter code. However, I am not aware if that is possible at all. Btw, TS explicitly recommends using modules over namespaces ([see](https://www.typescriptlang.org/docs/handbook/namespaces-and-modules.html)).

Aliasing the namespace name to a single letter is a way to imitate the direct interface imports from a module.

I do agree that having the alias requires the code reader to acknowledge it first, but in the same time it reduces the clutter significantly. Wdyt?

Comment (pan-kot) [reply src/property-filter/controller.ts]: Actually, I found a better way -- updated the PR, please take a look.

**Review by johannes-weber — APPROVED** [review]
LGTM.

**Comment by pan-kot** [comment]
Synced with main.

**Review by johannes-weber — APPROVED** [review]

**Pull request merged**